apps/total-typescript/src/pages/invoices/[merchantChargeId].tsx

```diff
@@ -37,38 +37,13 @@ export const getServerSideProps: GetServerSideProps = async ({
     if (merchantCharge && merchantCharge.identifier) {
       const charge = await stripe.charges.retrieve(merchantCharge.identifier)
 
-      const merchantProductId = merchantCharge.merchantProductId
-      const merchantProduct = await prisma.merchantProduct.findUnique({
-        where: {
-          id: merchantProductId as string,
-        },
-        select: {
-          productId: true,
-        },
-      })
-      const purchase = await prisma.purchase.findFirst({
-        where: {
-          merchantChargeId: merchantCharge.id,
-        },
-        select: {
-          bulkCouponId: true,
-        },
-      })
-      const bulkCouponId = purchase?.bulkCouponId
-      const bulkCoupon = bulkCouponId
-        ? await prisma.coupon.findUnique({
-            where: {
-              id: bulkCouponId,
-            },
-            select: {
-              maxUses: true,
-            },
-          })
-        : null
+      const purchase = await getPurchaseForStripeCharge(
+        merchantCharge.identifier,
+      )
+      const bulkCoupon = purchase && purchase.bulkCoupon
 
-      const productId = merchantProduct?.productId
       const product = await getProduct({
-        where: {id: productId},
+        where: {id: purchase?.productId},
```

Comment:
No need to Prisma query for the `merchantProduct` above, we can grab the `productId` off the `purchase` that we already queried for.